modules/widgets/wordpress-post-widget.php

```diff
@@ -27,138 +73,600 @@ public function __construct() {
+		}
+
+		return $widget_data;
+	}
+
+
+	/**
+	 * Gets blog data from the cache.
+	 *
+	 * @param string $site
+	 *
+	 * @return array|WP_Error
+	 */
+	public function get_blog_data( $site ) {
+		// load from cache, if nothing return an error
+		$site_hash = $this->get_site_hash( $site );
+
+		$cached_data = $this->wp_get_option( $this->widget_options_key_prefix . $site_hash );
+
+		/**
+		 * If the cache is empty, return an empty_cache error.
+		 */
+		if ( false === $cached_data ) {
+			return new WP_Error(
+				'empty_cache',
+				__( 'Information about this blog is being currently retrieved.', 'jetpack' )
+			);
+		}
+
+		return $cached_data;
+
+	}
+
+	/**
+	 * Fetch site information from the WordPress public API
+	 *
+	 * @param string $site URL of the site to fetch the information for.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function fetch_site_info( $site ) {
+
+		$response = $this->fetch_service_endpoint( sprintf( '/sites/%s', urlencode( $site ) ) );
+
+		return $response;
+	}
+
+	/**
+	 * Fetch list of posts from the WordPress public API.
+	 *
+	 * @param int $site_id The site to fetch the posts for.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function fetch_posts_for_site( $site_id ) {
+
+		$response = $this->fetch_service_endpoint(
+			sprintf(
+				'/sites/%1$d/posts/%2$s',
+				$site_id,
+				/**
+				 * Filters the parameters used to fetch for posts in the Display Posts Widget.
+				 *
+				 * @see    https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/
+				 *
+				 * @module widgets
+				 *
+				 * @since  3.6.0
+				 *
+				 * @param string $args Extra parameters to filter posts returned from the WordPress.com REST API.
+				 */
+				apply_filters( 'jetpack_display_posts_widget_posts_params', '' )
+			)
+		);
+
+		return $response;
+	}
+
+	/**
+	 * Parse external API response from the posts list request and handle errors if any occur.
+	 *
+	 * @param object|WP_Error $service_response The raw response to be parsed.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function parse_posts_response( $service_response ) {
+
+		/**
+		 * If the service returned an error, we pass it on.
+		 */
+		if ( is_wp_error( $service_response ) ) {
+			return $service_response;
+		}
+
+		/**
+		 * Check if the service returned proper posts array.
+		 */
+		if ( ! isset( $service_response->posts ) || ! is_array( $service_response->posts ) ) {
+			return new WP_Error(
+				'no_posts',
+				__( 'No posts data returned by remote.', 'jetpack' ),
+				'No posts information set in the returned data.'
+			);
+		}
+
+		/**
+		 * Format the posts to preserve storage space.
+		 */
+
+		return $this->format_posts_for_storage( $service_response );
+	}
+
+
+	/**
+	 * Parse external API response from the site info call and handle errors if they occur.
+	 *
+	 * @param array|WP_Error $service_response The raw response to be parsed.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function parse_site_info_response( $service_response ) {
```

Comment:
I reordered them in 9845e66c8ad293acf5145c9f790163ce33977b8d. I think they are better organized now.